File: SuperiorPackGroup/BLL/LocationTransfersBLL.vb
Imports DXDAL.SPGData
Imports DevExpress.Data.Filtering

Public Class LocationTransfersBLL

    Shared Function ValidateRecord(transferNumber As Integer?, fromLocationID As Integer?, toLocationID As Integer?, carrierID As Integer?, transferRecord As LocationTransfers, session As DevExpress.Xpo.Session) As Boolean

        If fromLocationID.HasValue = False OrElse toLocationID.HasValue = False OrElse carrierID.HasValue = False Then
            Return False
        End If

        If fromLocationID.Value = toLocationID.Value Then
            Return False
        End If

        If transferNumber.HasValue = False OrElse transferNumber.Value = 0 Then
            Dim maxTransferNumber As Integer? = CType(session.Evaluate(Of LocationTransfers)(New AggregateOperand("", "TransferNumber", Aggregate.Max), Nothing), Integer?)
            If maxTransferNumber.HasValue Then
                transferNumber = maxTransferNumber + 1
            Else
                transferNumber = 1
            End If

            transferRecord.TransferNumber = transferNumber.Value
        End If

        Return True

    End Function


End Class

File: SuperiorPackGroup/BLL/LocationTransfersBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using DXDAL.SPGData;
using DevExpress.Data.Filtering;

namespace SuperiorPackGroup
{
	public class LocationTransfersBLL
	{

		public static bool ValidateRecord(int? transferNumber, int? fromLocationID, int? toLocationID, int? carrierID, LocationTransfers transferRecord, DevExpress.Xpo.Session session)
		{

			if (fromLocationID.HasValue == false || toLocationID.HasValue == false || carrierID.HasValue == false)
			{
				return false;
			}

			if (fromLocationID.Value == toLocationID.Value)
			{
				return false;
			}

			if (transferNumber.HasValue == false || transferNumber.Value == 0)
			{
				int? maxTransferNumber = (int?)session.Evaluate<LocationTransfers>(new AggregateOperand("", "TransferNumber", Aggregate.Max), null);
				if (maxTransferNumber.HasValue)
				{
					transferNumber = maxTransferNumber + 1;
				}
				else
				{
					transferNumber = 1;
				}

				transferRecord.TransferNumber = transferNumber.Value;
			}

			return true;

		}


	}

}